luarules/gadgets/unit_shipyard_waterline_build.lua

```diff
@@ -0,0 +1,59 @@
+function gadget:GetInfo()
+	return {
+		name = "Shipyards Build at Waterline",
+		desc = "Move the build position depending on unit needs.",
+		author = "robert the pie",
+		layer = 0,
+		enabled = true,
+	}
+end
+
+if not gadgetHandler:IsSyncedCode() then return false end
+
+
+local shipyardsDefIDPads	= {}
+local shipyardsDefIDWaterlines	= {}
+local subDefIDs		= {}
+do
+	local shipyardsNamesToPads = {
+		corsy = 4,
+		corasy = 8,
+		armsy = 14,
+		armasy = 2,
+	}
+	for name, pad in pairs(shipyardsNamesToPads) do
+		if UnitDefNames[name] then
+			shipyardsDefIDPads[UnitDefNames[name].id] = pad
+			shipyardsDefIDWaterlines[UnitDefNames[name].id] = UnitDefNames[name].waterline
+		end
+	end
+end
+
+for unitDefID, unitDef in pairs(UnitDefs) do
+	if unitDef.waterline and unitDef.minWaterDepth and unitDef.waterline >= 30 then
```

Comment:
could check movedef, it has an explicit submarine bool

Reply:
`isSubmarine`, also make sure not to index the movedef without a nil check because immobiles/aircraft don't have one
```
local md = unitDef.moveDef
if md and md.isSubmarine then
```